src/ServiceStack.OrmLite.Oracle/OracleOrmLiteDialectProvider.cs

```diff
@@ -232,14 +247,28 @@ public override string GetQuotedValue(object value, Type fieldType)
                 var dateValue = (DateTime)value;
                 string iso8601Format = "yyyy-MM-dd";
                 string oracleFormat = "YYYY-MM-DD";
-                if (dateValue.ToString("yyyy-MM-dd HH:mm:ss.fff").EndsWith("00:00:00.000") == false)
+                if (dateValue.ToString("yyyy-MM-dd HH:mm:ss.fffffff").EndsWith("00:00:00.0000000") == false)
```

Comment:
I think the test for "is this a start-of-day?" can be more concisely expressed by following [this pattern](https://github.com/ServiceStack/ServiceStack.Text/blob/master/src/ServiceStack.Text/Common/DateTimeSerializer.cs#L376-387) from ServiceStack.Text repo.

Something like: `if (dateValue.TimeOfDay.Ticks != 0)` would avoid the intermediate operations of converting to string and parsing the intermediate result.

I think this should work for both `DateTime` and `DateTimeOffset` values.